app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorBlue.java

```diff
@@ -17,6 +19,31 @@
 public class ActivitySectorBlue extends BaseActivitySector {
 
     private ActivitySectorBlueBinding binding;
+    private RecyclerView.AdapterDataObserver mDataObserver = new RecyclerView.AdapterDataObserver() {
+
+        @Override
+        public void onChanged() {
+            super.onChanged();
+            checkEmpty();
+        }
+
+        @Override
+        public void onItemRangeInserted(int positionStart, int itemCount) {
+            super.onItemRangeInserted(positionStart, itemCount);
+            checkEmpty();
+        }
+
+        @Override
+        public void onItemRangeRemoved(int positionStart, int itemCount) {
+            super.onItemRangeRemoved(positionStart, itemCount);
+            checkEmpty();
+        }
+
+        void checkEmpty() {
```

Comment:
Define checkEmpty() as abstract method in BaseActivitySector. Next, override this method in each SectorActivity and use correct bindings.